nupic/research/frameworks/pytorch/modules/sgdw.py

```diff
@@ -0,0 +1,162 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2019, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+import torch
+from torch.optim import Optimizer
+
+
+class _RequiredParameter(object):
+    """Singleton class representing a required parameter for an Optimizer."""
+
+    def __repr__(self):
+        return "<required parameter>"
+
+
+required = _RequiredParameter()
+
+
+class SGDW(Optimizer):
+    r"""Implements SGDW algorithm.
+
+    from: https://github.com/pytorch/pytorch/blob/
```

Comment:
Just curious - is this copied as-is from PyTorch, or did you make modifications? Assuming yes, could you clarify in the comments what changes were made?  Could these changes be done as a subclass of the pytorch SGDW class instead of copying the whole thing?

Reply:
It is not yet implemented in pytorch. It is there as pull request, to be merged, so we can't pull it in.